core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -488,18 +489,21 @@ class LogSegment(val log: FileRecords,
    * Delete this log segment from the filesystem.
    */
   def delete() {
-    val deletedLog = log.delete()
-    val deletedIndex = index.delete()
-    val deletedTimeIndex = timeIndex.delete()
-    val deletedTxnIndex = txnIndex.delete()
-    if (!deletedLog && log.file.exists)
-      throw new IOException("Delete of log " + log.file.getName + " failed.")
-    if (!deletedIndex && index.file.exists)
-      throw new IOException("Delete of index " + index.file.getName + " failed.")
-    if (!deletedTimeIndex && timeIndex.file.exists)
-      throw new IOException("Delete of time index " + timeIndex.file.getName + " failed.")
-    if (!deletedTxnIndex && txnIndex.file.exists)
-      throw new IOException("Delete of transaction index " + txnIndex.file.getName + " failed.")
+    def closeable(delete: () => Unit, fileType: String, file: File): Closeable = new Closeable {
```

Comment:
Do we need to pass in fileType since the file name will have the type?

Reply:
I was trying to retain the previous format. If you think there's no value, I can remove it.

Reply:
Thanks. I think we can just remove it.